src/logic/actions/EditorActions.ts

```diff
@@ -0,0 +1,145 @@
+import {IRect} from "../../interfaces/IRect";
+import {Settings} from "../../settings/Settings";
+import {RectUtil} from "../../utils/RectUtil";
+import {EditorData} from "../../data/EditorData";
+import {CanvasUtil} from "../../utils/CanvasUtil";
+import {ISize} from "../../interfaces/ISize";
+import React from "react";
+import {IPoint} from "../../interfaces/IPoint";
+import {DrawUtil} from "../../utils/DrawUtil";
+import {PrimaryEditorRenderEngine} from "../render/PrimaryEditorRenderEngine";
+
+export class EditorActions {
+
+    // =================================================================================================================
+    // RENDER ENGINES
+    // =================================================================================================================
+
+    public static mountSupportRenderingEngine(activeLabelType: LabelType) {
+        switch (activeLabelType) {
+            case LabelType.RECTANGLE:
+                EditorModel.supportRenderingEngine = new RectRenderEngine(EditorModel.canvas);
+                break;
+            case LabelType.POINT:
+                EditorModel.supportRenderingEngine = new PointRenderEngine(EditorModel.canvas);
+                break;
+            case LabelType.POLYGON:
+                EditorModel.supportRenderingEngine = new PolygonRenderEngine(EditorModel.canvas);
+                break;
+            default:
+                EditorModel.supportRenderingEngine = null;
+                break;
+        }
+    };
+
+    public static swapSupportRenderingEngine(activeLabelType: LabelType) {
+        EditorActions.mountSupportRenderingEngine(activeLabelType);
+    };
+
+    public static mountRenderEngines(activeLabelType: LabelType) {
+        EditorModel.primaryRenderingEngine = new PrimaryEditorRenderEngine(EditorModel.canvas);
+        EditorActions.mountSupportRenderingEngine(activeLabelType);
+    }
+
+    // =================================================================================================================
+    // RENDER
+    // =================================================================================================================
+
+    public static fullRender() {
+        DrawUtil.clearCanvas(EditorModel.canvas);
+        EditorModel.primaryRenderingEngine.drawImage(EditorModel.image, EditorModel.imageRectOnCanvas);
+        EditorModel.primaryRenderingEngine.render(EditorActions.getEditorData());
+        EditorModel.supportRenderingEngine && EditorModel.supportRenderingEngine.render(EditorActions.getEditorData());
+    }
+
+    // =================================================================================================================
+    // GETTERS
+    // =================================================================================================================
+
+    public static getImageRect(image: HTMLImageElement): IRect | null {
+        if (!!image) {
```

Comment:
w zasadzie tutaj image'a powinieneś zawsze dostać, bo w argumentach nie jest podany jako opcjonalny, więc ten "if" jest trochę zastanawiający xD tam gdzie uzywasz tej metody to jakieś wrażliwe / bugogenne miejsce?

Reply:
i podobnie w innych metodach :)